pkg/core/redis/pool.go

```diff
@@ -35,3 +48,52 @@ func NewPool(config Configuration) *redis.Pool {
 		},
 	}
 }
+
+func newSentinelPool(config Configuration) *redis.Pool {
+	sntnl := &sentinel.Sentinel{
+		Addrs:      config.Sentinel.Addrs,
+		MasterName: config.Sentinel.MasterName,
+		Dial: func(addr string) (redis.Conn, error) {
+			timeout := 500 * time.Millisecond
+			c, err := redis.Dial(
+				"tcp",
+				addr,
+				redis.DialConnectTimeout(timeout),
+				redis.DialReadTimeout(timeout),
+				redis.DialWriteTimeout(timeout),
+			)
+			if err != nil {
+				return nil, err
+			}
+			return c, nil
+		},
+	}
+	return &redis.Pool{
+		MaxIdle:     3,
```

Comment:
Could you refactor a bit, so that the pool creation parameters is the same for sentinel and non-sentinel usage?

Reply:
Sure! Updated, see if it looks good?:)